app/src/main/java/mozilla/lockbox/store/DataStore.kt

```diff
@@ -76,53 +80,57 @@ open class DataStore(
             .addTo(compositeDisposable)
     }
 
-    // Warning: this is testing code.
-    // It's only called immediately after the user has pressed "Use Test Data".
-    fun resetSupport(support: DataStoreSupport) {
-        if (stateSubject.value != State.Unprepared) {
-            backend.reset()
+    open fun get(id: String): Observable<ServerPassword?> {
+        return list.map { items ->
+            items.findLast { item -> item.id == id }
         }
-        this.support = support
-        this.backend = support.createLoginsStorage()
-        // we shouldn't set the status of this to Unprepared,
-        // as we don't want to change any UI.
     }
 
     private fun touch(id: String) {
         if (!backend.isLocked()) {
             backend.touch(id)
-            updateList()
-        }
-    }
-
-    open fun get(id: String): Observable<ServerPassword?> {
-        return list.map { items ->
-            items.findLast { item -> item.id == id }
+                .asSingle(Dispatchers.Default)
+                .subscribe { _, _ ->
+                    updateList()
+                }
+                .addTo(compositeDisposable)
         }
     }
 
-    fun unlock() {
+    private fun unlock() {
```

Comment:
Conceptually, I'd like to be able `unlock` and then be able to wait for it to finish before launching `sync`, all from within `updateCredentials`.